### PR TITLE
bradl3yC - Remove unused reducer / actions related to 2346 form

### PR DESCRIPTION
**Pull request by bradl3yC**
## Description
During the initial development of the 2346 MDT form, custom redux logic was introduced.  As this work evolved, this work was largely left orphaned in the code base in place of using the built-in form reducer.  This PR aims to remove those orphaned actions/reducers/consts

## Acceptance criteria
- [x] - remove unused redux logic without breaking form functionality

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/2346/app-entry.jsx [hunk 1]
```diff
@@ -1,12 +1,10 @@
 import 'platform/polyfills';
 import startApp from 'platform/startup';
 import manifest from './manifest.json';
-import reducer from './reducers';
```
Comment (mr0sari0) [inline src/applications/disability-benefits/2346/app-entry.jsx:4]: I thought we still needed to import and include the form system's reducer in our `startApp`. Is that not the case?

File: src/applications/disability-benefits/2346/reducers/index.js [hunk 2]
```diff
@@ -1,8 +0,0 @@
-import { createSaveInProgressFormReducer } from 'platform/forms/save-in-progress/reducers';
-import formConfig from '../config/form';
-import form2346Reducer from './form2346Reducer';
-
-export default {
-  form: createSaveInProgressFormReducer(formConfig),
```
Comment (mr0sari0) [inline src/applications/disability-benefits/2346/reducers/index.js:6]: Similar to what I mentioned in `app-entry.jsx`

File: src/applications/disability-benefits/2346/app-entry.jsx (hunk 1 above)
Comment (bradl3yC) [reply src/applications/disability-benefits/2346/app-entry.jsx]: We do!  I already added it back in by the time you left this comment :-P Mistake on my part

**Review by maharielrosario — APPROVED** [review]
Thanks for adding the save-in-progress reducer back. This LGTM!

**Pull request merged**